src/interrupt/mod.rs

```diff
@@ -118,21 +118,21 @@ pub struct MsiIrqSourceConfig {
 
 pub trait InterruptSourceGroup: Send + Sync {
     /// Enable the interrupt sources in the group to generate interrupts.
-    fn enable(&self) -> Result<()> {
+    fn enable(&mut self) -> Result<()> {
         // Not all interrupt sources can be enabled.
         // To accommodate this, we can have a no-op here.
         Ok(())
     }
 
     /// Disable the interrupt sources in the group to generate interrupts.
-    fn disable(&self) -> Result<()> {
+    fn disable(&mut self) -> Result<()> {
         // Not all interrupt sources can be disabled.
         // To accommodate this, we can have a no-op here.
         Ok(())
     }
 
     /// Inject an interrupt from this interrupt source into the guest.
-    fn trigger(&self, index: InterruptIndex) -> Result<()>;
+    fn trigger(&mut self, index: InterruptIndex) -> Result<()>;
```

Comment:
I would argue that triggering an interrupt may likely not change the interrupt source group state, so I'd remove the mutability here.

Reply:
Yap, right, I'll update it.